client/src/app/site/pages/meetings/pages/assignments/modules/assignment-poll/components/assignment-poll-vote/assignment-poll-vote.component.ts

```diff
@@ -235,6 +235,76 @@ export class AssignmentPollVoteComponent extends BasePollVoteComponent<ViewAssig
         }
     }
 
+    public async submitVotes(users: ViewUser[]): Promise<void> {
+        let maxVotesAmount = 0;
+        let pollMaximum = 0;
+        let isListOpt = false;
+
+        if (!this.hasAlreadyVoted() && !(this.getVotingError() === `You do not have the permission to vote.`)) {
+            maxVotesAmount = this.getVotesCount();
+            pollMaximum = this.poll.max_votes_amount;
+        }
+        for (let user of users) {
+            if (
+                !this.hasAlreadyVoted(user) &&
+                !(this.getVotingError(user) === `You do not have the permission to vote.`)
+            ) {
+                if (this.poll.isMethodY && this.poll.max_votes_per_option > 1 && this.isErrorInVoteEntry()) {
+                    this.raiseError(this.translate.instant(`There is an error in your vote.`));
+                    return;
+                }
+                maxVotesAmount += this.getVotesCount(user);
+                pollMaximum += this.poll.max_votes_amount;
+            }
+        }
+        for (let option of this.poll.options) {
+            if (option.isListOption) {
+                isListOpt = true;
+            }
+        }
+        if ((this.poll.isMethodYNA && !isListOpt) || this.poll.isMethodYN) {
+            pollMaximum *= users.length;
+        }
+
+        const title = this.translate.instant(`Submit selection now?`);
+        const content =
+            this.translate.instant(`Your votes`) +
+            `: ${maxVotesAmount}/${pollMaximum}<br>` +
+            this.translate.instant(`Your decision cannot be changed afterwards.`);
+
+        const confirmed = await this.promptService.open(title, content);
+        if (confirmed) {
+            let value = this.voteRequestData[this.user.id].value;
+            if (!this.hasAlreadyVoted() && !(this.getVotingError() === `You do not have the permission to vote.`)) {
+                this.deliveringVote[this.user.id] = true;
+                this.cd.markForCheck();
+
+                const votePayload = {
+                    value: value,
+                    user_id: this.user.id
+                };
+                await this.sendVote(this.user.id, votePayload);
+            }
+            for (let user of users) {
+                value = this.voteRequestData[user.id].value;
+                if (
+                    !this.hasAlreadyVoted(user) &&
+                    !(this.getVotingError(user) === `You do not have the permission to vote.`)
+                ) {
+                    this.deliveringVote[user.id] = true;
+                    this.cd.markForCheck();
+
+                    const votePayload = {
+                        value: value,
+                        user_id: user.id
+                    };
+
+                    await this.sendVote(user.id, votePayload);
+                }
+            }
+        }
+    }
```

Comment:
This `submitVotes` method is also too long.

Also given that the gist of what the `submitVotes` methods do should be the same for all three poll types, why not try to move this function into the `base-poll-vote.component.ts`. If there are variations, you could encase them into their own helper functions that you can override in the subclasses.